tools/windows-node-installer/pkg/cloudprovider/azure/azurevm.go

```diff
@@ -0,0 +1,593 @@
+func createNIC(ctx context.Context, az *AzureProvider, vnetName, subnetName, nsgName, ipName, nicName string) (nic network.Interface, err error) {
+	subnet, err := getVirtualNetworkSubnet(ctx, az.subnetsClient, vnetName, subnetName)
+	if err != nil {
+		fmt.Errorf("failed to get subnet: %v", err)
+	}
+
+	ip, err := getPublicIP(ctx, az.ipClient, ipName)
+	if err != nil {
+		fmt.Errorf("failed to get ip address: %v", err)
+	}
+
+	nicParams := network.Interface{
+		Name:     to.StringPtr(nicName),
+		Location: to.StringPtr("centralus"),
+		InterfacePropertiesFormat: &network.InterfacePropertiesFormat{
+			IPConfigurations: &[]network.InterfaceIPConfiguration{
+				{
+					Name: to.StringPtr("ipConfig1"),
+					InterfaceIPConfigurationPropertiesFormat: &network.InterfaceIPConfigurationPropertiesFormat{
+						Subnet:                    &subnet,
+						PrivateIPAllocationMethod: network.Dynamic,
+						PublicIPAddress:           &ip,
+					},
+				},
+			},
+		},
+	}
+
+	if nsgName != "" {
+		nsg, err := getNetworkSecurityGroup(ctx, az.nsgClient, nsgName)
+		if err != nil {
+			fmt.Errorf("failed to get nsg: %v", err)
+		}
+		nicParams.NetworkSecurityGroup = &nsg
+	}
+
+	future, err := az.nicClient.CreateOrUpdate(ctx, resourceGroupName, nicName, nicParams)
+	if err != nil {
+		return nic, fmt.Errorf("cannot create nic: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, az.nicClient.Client)
+	if err != nil {
+		return nic, fmt.Errorf("cannot create nic: %v", err)
+	}
+
+	return future.Result(az.nicClient)
+}
+
+// createSecurityGroup tries to create a security group that allows to RDP to the windows node and allow for all the traffic
+// coming from the nodes that belong to the same VNET.
+func createSecurityGroup(ctx context.Context, az *AzureProvider, nsgName string) (err error) {
+	future, err := az.nsgClient.CreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		nsgName,
+		network.SecurityGroup{
+			Location: to.StringPtr("centralus"),
+			SecurityGroupPropertiesFormat: &network.SecurityGroupPropertiesFormat{
+				SecurityRules: &[]network.SecurityRule{
+					{
+						Name: to.StringPtr("rdp_access"),
+						SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+							Protocol:                 network.SecurityRuleProtocolTCP,
+							SourceAddressPrefix:      to.StringPtr("*"),
+							SourcePortRange:          to.StringPtr("1-65535"),
+							DestinationAddressPrefix: to.StringPtr("*"),
+							DestinationPortRange:     to.StringPtr("3389"),
+							Access:                   network.SecurityRuleAccessAllow,
+							Direction:                network.SecurityRuleDirectionInbound,
+							Priority:                 to.Int32Ptr(100),
+						},
+					},
+					{
+						Name: to.StringPtr("vnet_traffic"),
+						SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+							Protocol:                 network.SecurityRuleProtocolTCP,
+							SourceAddressPrefix:      to.StringPtr("10.0.0.0/16"),
```

Comment:
this should be obtained from Vnet range instead of hardcoding. See aws `ingress` function for example